tests/VirtoCommerce.NotificationsModule.Tests/UnitTests/NotificationMessageSearchServiceUnitTests.cs

```diff
@@ -0,0 +1,211 @@
+
+        SetupRepository(entities);
+
+        var searchCriteria = new NotificationMessageSearchCriteria
+        {
+            ObjectIds = [tenantId],
+            ObjectTypes = [tenantType],
+            Skip = 0,
+            Take = 10,
+        };
+
+        // Act
+        var result = await _searchService.SearchMessageAsync(searchCriteria);
+
+        // Assert
+        Assert.NotNull(result);
+        Assert.Single(result.Results);
+        Assert.Equal(tenantId, result.Results.First().TenantIdentity.Id);
+        Assert.Equal(tenantType, result.Results.First().TenantIdentity.Type);
+    }
+
+    [Fact]
+    public async Task SearchMessageAsync_ShouldFilterByNotificationType()
+    {
+        // Arrange
+        var notificationType = "RegistrationEmailNotification";
+
+        var entities = new List<NotificationMessageEntity>
+        {
+            new EmailNotificationMessageEntity { Id = Guid.NewGuid().ToString(), NotificationType = notificationType },
+        };
+
+        SetupRepository(entities);
+
+
+        var searchCriteria = new NotificationMessageSearchCriteria
+        {
+            NotificationType = notificationType,
+            Skip = 0,
+            Take = 10,
+        };
+        // Act
+        var result = await _searchService.SearchMessageAsync(searchCriteria);
+
+        // Assert
+        Assert.NotNull(result);
+        Assert.Single(result.Results);
+        Assert.Equal(notificationType, result.Results.First().NotificationType);
+    }
+
+    [Theory]
+    [InlineData("keyword1")]
+    [InlineData("keyword2")]
+    public async Task SearchMessageAsync_ShouldFilterByKeyword(string keyword)
+    {
+        // Arrange
+        var entities = new List<NotificationMessageEntity>
+        {
+            new EmailNotificationMessageEntity
+            {
+                Id = Guid.NewGuid().ToString(),
+                Subject = keyword,
+                Body = keyword,
+                From = keyword,
+                To = keyword,
+                CC = keyword,
+                BCC = keyword,
+                LastSendError = keyword,
+            },
+        };
+
+        SetupRepository(entities);
+
+        var searchCriteria = new NotificationMessageSearchCriteria
+        {
+            Keyword = keyword,
+            Skip = 0,
+            Take = 10,
+        };
+
+        // Act
+        var result = await _searchService.SearchMessageAsync(searchCriteria);
+
+        // Assert
+        var firstResult = (EmailNotificationMessage)result.Results.First();
+        Assert.NotNull(result);
+        Assert.Single(result.Results);
+        Assert.Contains(keyword, firstResult.Subject);
+        Assert.Contains(keyword, firstResult.Body);
+        Assert.Contains(keyword, firstResult.From);
+        Assert.Contains(keyword, firstResult.To);
+        Assert.Contains(keyword, firstResult.CC);
+        Assert.Contains(keyword, firstResult.BCC);
+        Assert.Contains(keyword, firstResult.LastSendError);
```

Comment:
This looks like all fields must contain the keyword, but it can only be in one of them.